src/controllers/tripController.js

```diff
@@ -0,0 +1,211 @@
+const { client } = require('../config/configDB')
+const jwt = require('jsonwebtoken');
+const JWT_SECRET = process.env.JWT_SECRET
+
+const createTrip = async (req, res, next) => {
+    const token = req.header('auth-token');
+    const { name, origin, destination, desc, departure_dateTime, arrival_dateTime } = req.body
+    const data = jwt.verify(token, JWT_SECRET);
+    const user_id = data.user.user_id;
+    try {
+        client.query(`
+        INSERT INTO trips (trip_name, trip_origin, trip_destination, trip_desc, trip_departure_datetime, trip_arrival_datetime)
+        VALUES ($1, $2, $3, $4, $5, $6)
+        RETURNING trip_id`,
+        [name, origin, destination, desc, departure_dateTime, arrival_dateTime],
+        (err, results) => {
+            if (err) {
+                throw err;
+            }
+            const trip_id = results.rows[0].trip_id;
+
+            // Step 3: Insert a record into the user_trip table
+            client.query(
+                `
+                INSERT INTO user_trip (trip_id, user_id)
+                VALUES ($1, $2)`,
+                [trip_id, user_id],
+                (err) => {
+                    if (err) {
+                        throw err;
+                    }
+                });
+            }
+        )
+            
+        const message = "Trip created successfully";
+        res.json({ message });
+    } catch (error) {
+        next(error);
+    }
+}
+
+const UpdateTrip = async (req, res, next) => {
+    const { name, origin, destination, desc, departure_dateTime, arrival_dateTime } = req.body
+    const trip_id = req.params.trip_id;
+    try {
+        client.query(`
+        
+        UPDATE trips SET trip_name = $2, trip_origin = $3, trip_destination = $4,trip_desc = $5, trip_departure_datetime = $6, trip_arrival_datetime = $7
+        WHERE trip_id = $1
+        `,[trip_id, name, origin, destination, desc, departure_dateTime, arrival_dateTime],(err,results)=>{
+            if(err)
+            {
+                throw err;
+            }
+        })
+        res.json({message: "trip updated successfully!"});
+    } catch (error) {
+        next(error);
+    }
+}
+
+const deleteTrip = async (req, res, next) => {
+    const trip_id = req.params.trip_id;
+    try {
+        client.query(`DELETE FROM trips
+        WHERE trip_id = $1`,[trip_id],(err, results) => {
+            if(err){
+                throw err;
+            }
+        })
+        res.json({message: "Trip deleted successfully."})
+    } catch (error) {
+        next(error);
+    }
+}
+
+const queryTripsByCommunityId = async (req, res, next) => {
```

Comment:
if we are using community id, the endpoint is probably /communities/:communityid/trips/:tripId
in which case I think this should be part of the communityController.js

Reply:
But we want that user can search for what their known users are planning for trips then they have asked us that user can query the result like filter by origin, destination and so on

Reply:
yeah, but this function is taking the query param from community id, so it should be a part of communityController as per the openapi spec

Reply:
Yeah you are correct but this is necessary as per the requirements

Reply:
The function is good, just change its location to the communityController. what is your concern?